pkgs/applications/terminal-emulators/kitty/default.nix

```diff
@@ -72,6 +72,10 @@ buildPythonApplication rec {
   # Causes build failure due to warning
   hardeningDisable = lib.optional stdenv.cc.isClang "strictoverflow";
 
+  preBuild = lib.optionalString (lib.versionAtLeast stdenv.hostPlatform.darwinMinVersion "11" && stdenv.isDarwin) ''
+    MACOSX_DEPLOYMENT_TARGET=10.16
```

Comment:
Does this has any effect? You are not using `export` here so it would be only accessible to bash but no child processes.

Reply:
All I can say is that without it the build doesn't work on my `aarch64-darwin` machine and there are other packages that use the same approach in `preBuild` and `preConfigure`. I don't know if the fact that this works is accidental or not.